src/materials/H5MoltresMaterial.C

```diff
@@ -0,0 +1,480 @@
+    _chi(declareProperty<std::vector<Real>>("chi")),
+    _chi_d(declareProperty<std::vector<Real>>("chi_d")),
+    _gtransfxs(declareProperty<std::vector<Real>>("gtransfxs")),
+    _beta_eff(declareProperty<std::vector<Real>>("beta_eff")),
+    _beta(declareProperty<Real>("beta")), // Why is this here
+    _decay_constant(declareProperty<std::vector<Real>>("decay_constant")),
+
+    _d_remxs_d_temp(declareProperty<std::vector<Real>>("d_remxs_d_temp")),
+    _d_fissxs_d_temp(declareProperty<std::vector<Real>>("d_fissxs_d_temp")),
+    _d_nsf_d_temp(declareProperty<std::vector<Real>>("d_nsf_d_temp")),
+    _d_fisse_d_temp(declareProperty<std::vector<Real>>("d_fisse_d_temp")),
+    _d_diffcoef_d_temp(declareProperty<std::vector<Real>>("d_diffcoef_d_temp")),
+    _d_recipvel_d_temp(declareProperty<std::vector<Real>>("d_recipvel_d_temp")),
+    _d_chi_d_temp(declareProperty<std::vector<Real>>("d_chi_d_temp")),
+    _d_chi_d_d_temp(declareProperty<std::vector<Real>>("d_chi_d_temp")),
+    _d_gtransfxs_d_temp(declareProperty<std::vector<Real>>("d_gtransfxs_d_temp")),
+    _d_beta_eff_d_temp(declareProperty<std::vector<Real>>("d_beta_eff_d_temp")),
+    _d_beta_d_temp(declareProperty<Real>("d_beta_d_temp")), // why is this here
+    _d_decay_constant_d_temp(declareProperty<std::vector<Real>>("d_decay_constant_d_temp")),
+    _interp_type(getParam<MooseEnum>("interp_type"))
+{
+  _num_groups = getParam<unsigned>("num_groups");
+  _num_precursor_groups = getParam<unsigned>("num_precursor_groups");
+  std::string base_file = getParam<std::string>("base_file");
+  _material_key = getParam<std::string>("material_key");
+  std::vector<std::string> xsec_names{"REMXS",
+                                      "FISSXS",
+                                      "NSF",
+                                      "FISSE",
+                                      "DIFFCOEF",
+                                      "RECIPVEL",
+                                      "CHI",
+                                      "CHI_D",
+                                      "GTRANSFXS",
+                                      "BETA_EFF",
+                                      "DECAY_CONSTANT"};
+
+  H5std_string FILE_NAME(base_file);
+  H5::H5File file_open(FILE_NAME, H5F_ACC_RDONLY);
+  if (!pathExists(file_open.getId(), _material_key + "/temp"))
+    mooseError("Unable to open database " + _material_key + "/temp");
+  H5std_string DATASET_NAME(_material_key + "/temp");
+  H5::DataSet dataset = file_open.openDataSet(DATASET_NAME);
+  H5::DataSpace filespace = dataset.getSpace();
+  int rank = filespace.getSimpleExtentNdims();
+  hsize_t dims[rank];
+  rank = filespace.getSimpleExtentDims(dims);
+  double data_out[dims[0]];
+  dataset.read(data_out, H5::PredType::NATIVE_DOUBLE);
+
+  _XsTemperature.resize(dims[0]);
+  for (int k = 0; k < dims[0]; ++k)
+  {
+    _XsTemperature[k] = data_out[k];
+  }
+
+  auto n = xsec_names.size();
+  for (decltype(n) j = 0; j < n; ++j)
```

Comment:
Since you are requiring C++17 in the makefile, you can make this shorter by changing "decltype(n)" to "auto".

Reply:
You're really going to require C++17? That honestly seems extreme since the gcc site itself only claims experimental support for this standard: https://gcc.gnu.org/projects/cxx-status.html#cxx17. Moreover, it looks gcc only reached full *experimental* support in gcc-8...which is pretty spanking new. I don't think this PR should be merged if C++17 is going to be required.

Reply:
Indeed we have not even been able to release a gcc-8 package yet for MOOSE

Reply:
my Makefile is wrong. I did not mean to require c++17.

Reply:
Well on third thought, the `-std=c++17` flag has been supported since gcc 5, although the majority of features were not added until gcc-7. What 17 features are you using? I guess you can take the mentality that if your users cannot compile Moltres, they should upgrade their compiler. If Moltres users are using the newest MOOSE package, they will have all the experimental 17 support gcc supplies outside of some fixups for deleting class template argument deductions...so maybe this is ok. I would like not to suppress exploration of new C++ features.